op-proposer/proposer/driver.go

```diff
@@ -296,6 +295,13 @@ func (l *L2OutputSubmitter) FetchDGFOutput(ctx context.Context) (*eth.OutputResp
 		return nil, false, fmt.Errorf("could not fetch output at current block number %d: %w", currentBlockNumber, err)
 	}
 
+	if claim == output.StateRoot {
```

Comment:
This is the wrong comparison. The `StateRoot` is the world state root but the proposal is actually the `output.OutputRoot`.